apps/asynchromix/asynchromix.py

```diff
@@ -0,0 +1,542 @@
+"""
+Implementation of Asynchromix MPC Coordinator using an EVM blockchain
+"""
+from web3 import Web3, HTTPProvider
+import time
+import asyncio
+import logging
+from contextlib import contextmanager
+import subprocess
+# from honeybadgermpc.exceptions import HoneyBadgerMPCError
+from honeybadgermpc.router import SimpleRouter
+from honeybadgermpc.utils.misc import subscribe_recv
+from honeybadgermpc.utils.misc import wrap_send
+from honeybadgermpc.field import GF
+from honeybadgermpc.polynomial import EvalPoint, polynomials_over
+from honeybadgermpc.elliptic_curve import Subgroup
+from honeybadgermpc.offline_randousha import randousha
+from honeybadgermpc.offline_randousha import generate_triples
+from honeybadgermpc.offline_randousha import generate_bits
+from honeybadgermpc.mpc import Mpc
+from honeybadgermpc.progs.mixins.share_arithmetic import BeaverMultiplyArrays
+from honeybadgermpc.progs.mixins.constants import MixinConstants
+from honeybadgermpc.progs.mixins.base import MixinBase
+from butterfly_network import iterated_butterfly_network
+# from math import log
+from ethereum.tools._solidity import compile_code as compile_source
+from web3.contract import ConciseContract
+import os
+
+
+# For fake preprocessing
+from honeybadgermpc.preprocessing import (
+    PreProcessedElements, clear_preprocessing)
+
+field = GF(Subgroup.BLS12_381)
+
+
+async def wait_for_receipt(w3, tx_hash):
+    while True:
+        tx_receipt = w3.eth.getTransactionReceipt(tx_hash)
+        if tx_receipt is not None:
+            break
+        await asyncio.sleep(5)
+    return tx_receipt
+
+
+def print_exception_callback(future):
```

Comment:
This could probably be placed in the `utils` files.

Reply:
fixed, also changed the reference in test_hbavss which is the only other place it's used